pkg/cache/cache_from.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"fmt"
+
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/okteto/okteto/pkg/registry"
+)
+
+func AddDefaultPullCache(b *model.BuildInfo) {
+	// Extract the registry and image name from the original Docker image
+	imageCtrl := registry.NewImageCtrl(okteto.Config{})
+	_, repo := imageCtrl.GetRegistryAndRepo(b.Image)
+	image, _ := imageCtrl.GetRepoNameAndTag(repo)
+
+	// Construct the cache image tags
```

Comment:
We should only add the global registry whenever the user has access to it. [registry.HasGlobalAccess](https://github.com/okteto/okteto/blob/master/pkg/registry/registry.go#L141) is your friend here 😄

Reply:
Done 👍